atomix/cluster/src/main/java/io/atomix/raft/impl/ReconfigurationHelper.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.atomix.raft.impl;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.cluster.messaging.MessagingException.NoRemoteHandler;
+import io.atomix.cluster.messaging.MessagingException.NoSuchMemberException;
+import io.atomix.raft.RaftError;
+import io.atomix.raft.RaftException.ProtocolException;
+import io.atomix.raft.RaftServer;
+import io.atomix.raft.RaftServer.Role;
+import io.atomix.raft.cluster.RaftMember;
+import io.atomix.raft.cluster.RaftMember.Type;
+import io.atomix.raft.cluster.impl.DefaultRaftMember;
+import io.atomix.raft.impl.RaftContext.State;
+import io.atomix.raft.protocol.JoinRequest;
+import io.atomix.raft.protocol.LeaveRequest;
+import io.atomix.raft.protocol.RaftResponse.Status;
+import io.atomix.raft.protocol.TransferRequest;
+import io.atomix.utils.concurrent.ThreadContext;
+import io.atomix.utils.logging.ContextualLoggerFactory;
+import io.atomix.utils.logging.LoggerContext;
+import java.net.ConnectException;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+
+public final class ReconfigurationHelper {
+
+  private final ThreadContext threadContext;
+  private final RaftContext raftContext;
+  private final Logger logger;
+
+  public ReconfigurationHelper(final RaftContext raftContext) {
```

Comment:
:wrench: If you plan on unit testing this, then I would not depend directly on the Raft context, it's too complex to stub or mock.

Reply:
I tried to not directly use it :sob:. But it is too dependent on `RaftContext`. It need access to runtime info - like who is the leader, trigger transition etc.